packages/app-store/googlecalendar/lib/CalendarService.ts

```diff
@@ -369,57 +370,75 @@ export default class GoogleCalendarService implements Calendar {
-    const key = JSON.stringify({ timeMin, timeMax, items });
-    const cached = await prisma.calendarCache.findUnique({
-      where: {
-        credentialId_key: {
-          credentialId: this.credential.id,
-          key,
+
+      freeBusyResult = apires.data;
+    } else {
+      const { timeMin: _timeMin, timeMax: _timeMax, items } = args;
+      const { timeMin, timeMax } = handleMinMax(_timeMin, _timeMax);
+      const key = JSON.stringify({ timeMin, timeMax, items });
+      const cached = await prisma.calendarCache.findUnique({
+        where: {
+          credentialId_key: {
+            credentialId: this.credential.id,
+            key,
+          },
+          expiresAt: { gte: new Date(Date.now()) },
         },
-        expiresAt: { gte: new Date(Date.now()) },
-      },
-    });
+      });
 
-    if (cached) return cached.value as unknown as calendar_v3.Schema$FreeBusyResponse;
+      if (cached) {
+        freeBusyResult = cached.value as unknown as calendar_v3.Schema$FreeBusyResponse;
+      } else {
+        const apires = await calendar.freebusy.query({
+          requestBody: { timeMin, timeMax, items },
+        });
 
-    const apires = await calendar.freebusy.query({
-      requestBody: { timeMin, timeMax, items },
-    });
+        // Skipping await to respond faster
+        await prisma.calendarCache.upsert({
+          where: {
+            credentialId_key: {
+              credentialId: this.credential.id,
+              key,
+            },
+          },
+          update: {
+            value: JSON.parse(JSON.stringify(apires.data)),
+            expiresAt: new Date(Date.now() + CACHING_TIME),
+          },
+          create: {
+            value: JSON.parse(JSON.stringify(apires.data)),
+            credentialId: this.credential.id,
+            key,
+            expiresAt: new Date(Date.now() + CACHING_TIME),
+          },
+        });
 
-    // Skipping await to respond faster
-    await prisma.calendarCache.upsert({
-      where: {
-        credentialId_key: {
-          credentialId: this.credential.id,
-          key,
-        },
-      },
-      update: {
-        value: JSON.parse(JSON.stringify(apires.data)),
-        expiresAt: new Date(Date.now() + CACHING_TIME),
-      },
-      create: {
-        value: JSON.parse(JSON.stringify(apires.data)),
-        credentialId: this.credential.id,
-        key,
-        expiresAt: new Date(Date.now() + CACHING_TIME),
-      },
-    });
+        freeBusyResult = apires.data;
+      }
+    }
+    if (!freeBusyResult.calendars) return null;
 
-    return apires.data;
+    const result = Object.values(freeBusyResult.calendars).reduce((c, i) => {
+      i.busy?.forEach((busyTime) => {
+        c.push({
+          start: busyTime.start || "",
+          end: busyTime.end || "",
+        });
+      });
+      return c;
+    }, [] as Prisma.PromiseReturnType<CalendarService["getAvailability"]>);
+    return result;
   }
 
   async getAvailability(
```

Comment:
Unsure how to write a test for this. @joeauyeung Do we already have tests for google calendar

Reply:
We don't have any merged yet but it's in the works.